rclcpp_examples/src/services/add_two_ints_client.cpp

```diff
@@ -35,6 +36,41 @@ example_interfaces::srv::AddTwoInts_Response::SharedPtr send_request(
   }
 }
 
+// This function could be considered a prototype of an API that should be
+// promoted into rclcpp. This version polls; doing it in an event-driven
+// manner would be better.
+bool wait_for_service(rclcpp::Node::SharedPtr node,
+  const std::string & topic,
+  const std::chrono::duration<float> timeout)
+{
+  const std::chrono::milliseconds sleep_per_loop(1);
+  auto end = std::chrono::steady_clock::now() + timeout;
+  bool success = false;
+  while (std::chrono::steady_clock::now() < end) {
+    try {
+      if (node->count_subscribers(topic) == 0) {
+        success = true;
+        break;
+      }
+    } catch (std::runtime_error & e) {
+      // fastrtps doesn't yet implement rmw_count_subscribers(), but rather
+      // throws an exception instead. We'll just run out the clock and
+      // return true, hoping for the best.
+      rmw_reset_error();
+      success = true;
+    }
```

Comment:
This is a fairly blanket error case, maybe you could either check the error message contents or at least print it out so that when there is an error that doesn't match what you expected you can see it in the log? That might break your example test though.